### ISSUE TITLE
Referencing ClassTableChild build table error

### ISSUE THREAD

**Issue by jmai00** [typeorm/typeorm#159]
I get a schema creation error when I put a OneToOne or OneToMany column that references a ClassTableChild.

Example Code is below:
```typescript
import {Table, Column, PrimaryGeneratedColumn, TableInheritance, DiscriminatorColumn} from "../src/index";

@Table()
@TableInheritance("class-table")
@DiscriminatorColumn({ name: "type", type: "string"})
export class PersonA {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    firstName: string;

    @Column()
    lastName: string;
}
```

```typescript
import {ClassTableChild, Column} from "../src/index";
import {Person} from "./Person";

@ClassTableChild()
export class Employee extends Person {
    @Column()
    salary: number;
}
```

```typescript
import {Table, OneToOne, JoinColumn, PrimaryGeneratedColumn} from "../src/index";
import {Employee} from "./Employee";

@Table()
export class Department {

    @PrimaryGeneratedColumn()
    id: number;

    @OneToOne( (type) => Employee, { cascadeInsert: true, nullable: false})
    @JoinColumn()
    manager: Employee;
}
```

```typescript
import {Table, JoinColumn, Column, PrimaryGeneratedColumn} from "../src/index";
import {OneToOne} from "../src/decorator/relations/OneToOne";
import {Employee} from "./Employee";

@Table()
export class Department {
    @PrimaryGeneratedColumn()
    id: number;
    
    @Column()
    name: string;

    @OneToOne( (type) => Employee, { cascadeInsert: true, nullable: false})
    @JoinColumn()
    manager: Employee;
}
```

```typescript
require("reflect-metadata");
import {createConnection, ConnectionOptions, Connection} from "./src/index";
import {Person} from "./entities/Person";
import {Department} from "./entities/Department";
import {Employee} from "./entities/Employee";

let dbConnection = {
    type: "mysql",
    host: "localhost",
    port: 3306,
    username: "root",
    password: "root",
    database: "test_dev",
};

createConnection({
    driver: dbConnection,
    entities: [
        Person, Employee, Department,
    ],
    dropSchemaOnConnection: true,
    autoSchemaSync: true,
} as ConnectionOptions).then( async (connection: Connection) => {
    let x = connection.entityManager;

    let department = new Department();
    department.name = "Software";

    let employee = new Employee();
    employee.firstName = "Hello";
    employee.lastName = "World";
    employee.salary = 1;
   
    department.manager = employee;

    await x.persist(department);

    console.log("Done!");
}).catch( (error) => {
    console.log(`Error: ${error.message}`);
});
```

Error:

> Cannot read property 'propertyType' of undefined

I traced this back to line 343 of EntityMetadataBuilder.ts where it's looking for the inverseSideColumn which is undefined in the relation.relation.joinColumn.referencedColumn. I assume the referenceColumn  is supposedly the created <superclass>Id column in the classtablechild

If I add a PrimaryGeneratedColumn into Employee Class, this error goes away.

**Comment by pleerock** [comment]
Yes, it was a bug and I have fixed that and released in `0.0.7-alpha.3`. Do `npm i typeorm@0.0.7-alpha.3`

**Comment by jmai00** [comment]
There is another issue which wasn't checked. After persisting, the classtablechild object (manager, Employee) doesn't get populate w/ the new database id. The department object does get the populated database id.

**Comment by pleerock** [comment]
fixed and released in `0.0.7-alpha.11`
